lib/query.js

```diff
@@ -1127,9 +1123,7 @@ Query.prototype.findOne = function (conditions, fields, options, callback) {
 
   if (options) {
     this.setOptions(options);
-  }
-
-  if (fields) {
+  } else if (fields) {
```

Comment:
Yeah the other two ifs are redundant, you're right. Just remove the L-1126 change and this is good :)